### DISCUSSION TITLE
Where are the flux command created resources stored?

### DISCUSSION THREAD

**Discussion by peterb-vetct** [fluxcd/flux2#3504] [General] [answered]
Hi All,

When I create a new resource with th _flux_ command (and not via manifest file), where is this resource configuration stored?
To be clear, I run this command 
`flux create kustomization thisKustomization --source=thisGitRepo --path=./thisSubPath --prune=true --interval=5m ` 
flux creates the new kustomization configuration and works perfectly. But, where is this configuration stored? If I restart kustomization-controller the kustomization configuration is still there, but there is no persistent volume or any cache/db used.
I was reading documentation and searching on the internet, but no luck to figure it out. 
Someone, please enlighten me.
Many thanks,

**Comment by makkes** [comment, marked as answer]
The Kustomization is a [custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) created from a custom resource definition (CRD). It is stored in Kubernetes' backing store (etcd). You can do a `kubectl -n flux-system get kustomization thisKustomization` to retrieve it; append `-o yaml` to retrieve the complete manifest.

**Reply by peterb-vetct** [reply]
That makes sense, thank you for your answer.
I have a second question regarding this. What is the preferred way to manage the custom resource on the prodcution setup? ~~Store~~ Manage them in etcd or as a manifest in git?

**Reply by makkes** [reply]
Please see https://fluxcd.io/flux/concepts/ for an introduction of GitOps as a concept and https://fluxcd.io/flux/get-started/ for getting started with Flux for implementing GitOps.
